### ISSUE TITLE
Reactables should preserve stream type

### ISSUE THREAD

**Issue by danschultz** [danschultz/frappe#17]
The following code is throwing a `Uncaught Error: Bad state: Stream has already been listened to.` error:

```dart
var searchInput = $["search-input"] as TextInputElement;

var onInput = new EventStream(searchInput.onChange)
    .debounce(new Duration(microseconds: 250))
    .map((_) => searchInput.value)
    .distinct();

onInput.forEach((search) => print("searching for $search"));

var suggestions = onInput.flatMapLatest((search) => _queryMovies(search).asStream());

suggestions.forEach((suggestions) => print(suggestions));
```

Since `searchInput.onChange` is a broadcast stream, I'd expect that I'd be able to add multiple listeners to the reactable transformation.

**Comment by Andersmholmgren** [comment]
FYI I just attempted to upgrade to 0.3.2 but I hit this issue all over the place

**Comment by danschultz** [comment]
Thanks for the report. I'll tackle this next.

**Comment by danschultz** [comment]
These issues should be fixed in the [v0.4](https://github.com/danschultz/frappe/tree/v0.4) branch
